IBPSA/Electrical/BaseClasses/PV/PVOpticalAbsRat.mo

```diff
@@ -0,0 +1,214 @@
+
+  extends IBPSA.Electrical.BaseClasses.PV.BaseClasses.PartialPVOptical;
+
+ parameter Modelica.Units.SI.Height alt
+   "Site altitude in Meters, default= 1"
+   annotation(Dialog(group="Location"));
+
+ parameter Real groRef(unit="1") "Ground reflectance"
+   annotation ();
+
+  constant Modelica.Units.SI.Irradiance HGloHor0=1000
+ "Total solar radiation on the horizontal surface under standard conditions"
+  annotation(Dialog(group="Location"));
+
+  constant Modelica.Units.SI.Irradiance G_sc=1376 "Solar constant";
+
+  parameter Real glaExtCoe(unit="1/m")=4
+  "Glazing extinction coefficient for glass";
+
+  parameter Modelica.Units.SI.Length glaThi=0.002
+  "Glazing thickness for most PV cell panels it is 0.002 m";
+
+  parameter Real refInd(unit="1")=1.526
+  "Effective index of refraction of the cell cover (glass)";
+
+  Modelica.Units.SI.Irradiance HDirHor
+   "Beam solar radiation on the horizontal surface";
+
+  Modelica.Units.SI.Angle incAngGro "Incidence angle for ground reflection";
+
+  Modelica.Units.SI.Angle incAngDif "Incidence angle for diffuse radiation";
+
+  Real incAngMod(final unit="1", min=0) "Incidence angle modifier";
+
+  Real incAngModGro(final unit="1", min=0)
+  "Incidence angle modifier for ground refelction";
+
+  Real incAngModDif(final unit="1", min=0)
+  "Incidence angle modifier for diffuse radiation";
+
+  Modelica.Units.SI.Angle refAng "Angle of refraction";
+
+  Modelica.Units.SI.Angle refAngGro "Angle of refraction for ground reflection";
+
+  Modelica.Units.SI.Angle refAngDif
+    "Angle of refraction for diffuse irradiation";
+
+  Real tau(final unit="1", min=0)
+  "Transmittance of the cover system";
+
+  Real tau_gro(final unit="1", min=0)
+  "Transmittance of the cover system for ground reflection";
+
+  Real tau_dif(final unit="1", min=0)
+    "Transmittance of the cover system for diffuse radiation";
+
+  Real R_b(final unit="1", min=0)
+   "Ratio of irradiance on tilted surface to horizontal surface";
+
+  Modelica.Units.SI.Angle zen "Zenith angle";
+
+  BaseClasses.PVOptical.AirMass airMas(final alt=alt) "Air mass computation"
+    annotation (Placement(transformation(extent={{-60,60},{-40,80}})));
+  BaseClasses.PVOptical.AirMassModifier airMasMod(final PVTecTyp=PVTecTyp)
+    "Air mass modifier computation depending on PV type"
+    annotation (Placement(transformation(extent={{20,60},{40,80}})));
+  Modelica.Blocks.Interfaces.RealOutput absRadRat(final unit="1")
+    "Ratio of absorbed radiation under operating conditions to standard conditions"
+    annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+
+  //Conditional connectors
+
+protected
+  parameter Real tau_0(final unit="1")=exp(-(glaExtCoe*glaThi))*
+          (1 - ((refInd - 1)/(refInd + 1))^2)
+      "Transmittance at standard conditions (incAng=refAng=0)";
+
+
+equation
+// Restriction for zenith angle
+zen = if zenAng <= Modelica.Constants.pi/2 then
+zenAng
+ else
+   Modelica.Constants.pi/2;
```

Comment:
Please go through all the code and make sure it is properly formatted so it is easier to read for users and other developers.

Reply:
done